src/5_banner.js

```diff
@@ -228,6 +228,18 @@ banner = function(branch, options, linkData, storage) {
 			};
 		}
 
+		var modalBackground = doc.getElementById('branch-banner-modal-background');
+
+		if (modalBackground) {
+			modalBackground.onclick = function(ev) {
```

Comment:
avoid duplicating the code https://github.com/BranchMetrics/web-branch-deep-linking-attribution/blob/4835ce92e197b314baa4857e96de49e72edd1c69/src/5_banner.js#L222 

Have a common event handler function like 

```
function closeBanner(ev) {
				ev.preventDefault();
				branch._publishEvent('willCloseBanner');
				closeBanner({}, function() {
					branch._publishEvent('didCloseBanner');
				});
			};
closeButton.addEventListener('click',closeBanner);
modalBackground.addEventListener('click',closeBanner);
```

Reply:
@nsingh-branch would you want this new function live top-level, or should it be defined inside of other functions? (in this case, this code logic is used within finalHookupsCallback, so I could have it live within finalHookupsCallback, or within the banner function)

Reply:
@yli-branch Yes within finalHookupsCallback like my above code.
btw you are tagging a wrong reviewer here. ( nsingh-branch) is not the reviewer here.

Reply:
thanks for the callout!